api/v1beta2/cloudstackmachinestatechecker_types.go

```diff
@@ -0,0 +1,59 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta2
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// CloudStackMachineStateCheckerSpec
+type CloudStackMachineStateCheckerSpec struct {
```

Comment:
do we plan to have any concept of "FailureDomainHealthChecker"? Where if a failure domain is down, we move away from it to another one?

Reply:
I certainly like the idea, but that comes to a much bigger discussion which may even involve CAPI implementation requirements.

We've kicked around the idea before. : )